src/Microsoft.Extensions.ApiDescription.Client/build/Microsoft.Extensions.ApiDescription.Client.targets

```diff
@@ -279,8 +296,17 @@
       unique.
     -->
     <ItemGroup>
-      <Compile Remove="@(ServiceFileReference -> '%(OutputPath)')" />
-      <Compile Include="@(ServiceFileReference -> '%(OutputPath)')" SourceDocument="%(FullPath)" />
+      <TypeScriptCompile Remove="@(ServiceFileReference -> '%(RelativeOutputPath)')" />
+      <TypeScriptCompile Include="@(ServiceFileReference -> '%(RelativeOutputPath)')"
```

Comment:
Can we just do this based on the file extensions of the files? Then we don't need nearly as much plumbing.

Reply:
Are you suggesting changing the `Condition`s to something like `'%(ServiceFileReference.Extension)' == '.ts'` and `'%(ServiceFileReference.Extension)' != '.ts'` and removing the `%(TargetLanguage)` metadata? That's possible.

My main concern w.r.t. extensions is users have complete control of the generated filenames. Put another way, their `%(CodeGenerator)` choices are more reliable than `%(OutputPath)`.

Reply:
But if I generate files as `.cs.txt` and you don't add them to the `Compile` group... that sounds great to me. Put another way, I think users experience is that the project file automatically includes or excludes files based on their extension - it's something that any users expects.

Reply:
The condition could be `'%(ServiceFileReference.Extension)' == '.cs'` or `'%(ServiceFileReference.Extension)' == '.cs' OR '%(ServiceFileReference.Extension)' == '.vb'` and so on for all languages that we know use `@(Compile)` items. Which are you suggesting?

If the second, is there a canonical list somewhere (I haven't found one) listing languages, their extensions, and the items they use?

Reply:
I'm suggesting that we'd start with those conditions for `cs`, `vb, `fs`, and `ts` because it's trivial to do. There's examles of code like this in the .NET SDK